README.md

```diff
@@ -198,6 +199,15 @@ A `TraceApi` object is guaranteed to be returned by both of these calls, even if
 
 A fully detailed overview of the `TraceApi` object is available [here](doc/trace-api.md).
 
+## How does automatic tracing work?
+
+Node.js provides concurrency through an asynchronous execution model. To be able to keep track of the current request across asynchronous execution boundaries we use the [async-listener][] module. It modifies the core async APIs in Node.js, and Promises, to save and restore the 'execution context' across the async bounaries. In addition, we instrument frameworks (like express, hapi, etc. if loaded) to create request contexts for the execution of the incoming requests. Similarly, we instrument RPC libraries (like mysql, redis, outbound http, etc.), utilizing the saved/restored context, to keep track of the latency of these operations and to correlate them to the incoming request.
+
+This `async-listener` based mechanism works great in most cases, but it does have some limitations that can prevent us from being able to properly track execution context:
+
+* It is possible to use JavaScript code that does its own queuing of callback functions – simulating async hops. For example, one may write a http request buffering library that queues requests and then performs them in a batch in one shot. In such a case, when all the callbacks fire, they seem to be executing on behalf of the wrong incoming context. This problem is called the pooling problem or the [user-space queuing problem][queueing-problem], and is a fundamental limitation of JavaScript. If your application uses such code, you will notice that certain portions of your outbound RPCs do not get traced. In such cases we try to work around the problem through monkey patching, or work with the library authors to fix the code to properly propagate context. However, finding problematic code is not always trivial.
```

Comment:
First sentence: maybe "simulating async hops" can be removed or replaced with "effectively merging asynchronous execution contexts"?

Third sentence: "they seem to be executing on behalf of the wrong incoming context" -> "they will execute in the context which flushed the queue instead of the contexts which added the callbacks to the queue". In practice that context usually ends up being correct for one of the callbacks just incorrect for all others (depending on the exact flushing policy).

Fifth sentence: "you will notice that RPCs from many requests are showing up under a single trace or that certain portions of your outbound RPCs do not get traced"